src/lib/translators/mongo.ts

```diff
@@ -98,6 +105,34 @@ function transformSort(step: SortStep): MongoStep {
   return { $sort: sortMongo };
 }
 
+/** transform an 'top' step into corresponding mongo steps */
+function transformTop(step: TopStep): Array<MongoStep> {
+  const sortOrder = step.sort === 'asc' ? 1 : -1;
+  const groupMongo: MongoStep = {};
+  let groupCols: PropMap<string> | null = {};
+
+  // Prepare the $group Mongo step
+  if (step.groups) {
+    for (const col of step.groups) {
+      groupCols[col] = `$${col}`;
+    }
+  } else {
+    groupCols = null;
+  }
+  groupMongo['$group'] = {
+    _id: groupCols,
+    _tcAppArray: { $push: '$$ROOT' },
+  };
+
+  return [
+    { $sort: { [step.value]: sortOrder } },
+    groupMongo,
```

Comment:
This is purely a matter of style but I wonder if it would not be more readable to avoid an intermediate `groupMongo` variable and let the step be explicit:

```suggestion
    { $group: { _id: groupCols, _tcAppArray: { $push: '$$ROOT' } } },
```